core/primitives/src/views.rs

```diff
@@ -1000,6 +1001,32 @@ impl From<ExecutionStatus> for ExecutionStatusView {
     }
 }
 
+#[derive(BorshSerialize, BorshDeserialize, Serialize, Deserialize, PartialEq, Clone, Eq, Debug)]
+pub struct ProfileDataView {
+    costs: Vec<(Cost, u64)>,
+}
```

Comment:
Hm, the JSON serialization doesn't look right to me:

```rust
#[test]
fn profile_view_json_repr() {
    let pdv = ProfileDataView { costs: vec![(Cost::ExtCost(ExtCosts::write_memory_base), 92)] };
    let json = serde_json::to_string(&pdv).unwrap();
    assert_eq!(&json, r#"{"costs":[[{"ExtCost":"write_memory_base"},92]]}"#)
}
```

I think what we want to see here is just `{"write_memory_base": 92}`. The current representation includes a bunch of details about how our code is structured today. But we should base the JSON repr on what the actual data are we storing, and how to present the data to the user. The data is a *mapping* of costs to values, it's not an array with pairs of objects. 

I guess it probably makes sense to serialize u64 as string, as we do for gas throughout the API.

Reply:
I particularly considered `{"write_memory_base": 92}` form, but it's losing the of ExtCost type information. I think it's worth given future add new type of cost other than ActionCosts and ExtCosts, there're risks of collision. Also, A clear glance to separate action cost and ext cost is important to help debugging cost difference